### PR TITLE
Notification switch fix

### PR DESCRIPTION
**Pull request by gbielanski**
please review

### CODE REVIEW

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java [hunk 1]
```diff
@@ -82,16 +83,18 @@ protected boolean isMarkedForNotification() {
...
-    protected void markForNotification(View view) {
+    protected void markForNotification(CompoundButton buttonView, boolean isChecked) {
         SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(currentSector());
         SharedPreferences.Editor editor = sharedPreferences.edit();
-        editor.putInt(NotificationUtils.NOTIFICATION_SECTOR_COLOR, SectorColor.toInt(sectorColor()));
-        editor.apply();
-        if (((Switch) view).isChecked())
+
+        if (isChecked) {
+            editor.putInt(NotificationUtils.NOTIFICATION_SECTOR_COLOR, SectorColor.toInt(sectorColor()));
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Extract this editor.putInt(...) into separate method. Pass sectorColor() as parameter. Use here and in else statement.

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java [hunk 2]
```diff
@@ -82,16 +83,18 @@ protected boolean isMarkedForNotification() {
...
             Toast.makeText(currentSector(), getString(R.string.setup_notification, sectorColor().toString()), Toast.LENGTH_LONG).show();
```
Comment (miltomasz) [inline app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Extract toasts into separate method. Pass R.string.(...) as parameters.

**Review by miltomasz — CHANGES_REQUESTED** [review]

**Comment by miltomasz** [comment]
Once you're done with these changes you can merge it.

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java (hunk 1 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Changed

File: app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java (hunk 2 above)
Comment (gbielanski) [reply app/src/main/java/pl/example/android/garbageapp/ui/BaseActivitySector.java]: Changed

**Comment by gbielanski** [comment]
Please review latest version

**Comment by gbielanski** [comment]
Please review color change.

**Review by miltomasz — APPROVED** [review]

**Pull request merged**